internal/controller/postgrescluster/pgadmin_test.go

```diff
@@ -604,6 +606,136 @@ volumeMode: Filesystem
 	})
 }
 
+func TestReconcilePGAdminUsers(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("Disabled", func(t *testing.T) {
+		r := new(Reconciler)
+		cluster := new(v1beta1.PostgresCluster)
+		assert.NilError(t, r.reconcilePGAdminUsers(ctx, cluster, nil, nil))
+	})
+
+	// pgAdmin enabled
+	cluster := &v1beta1.PostgresCluster{}
+	cluster.Namespace = "ns1"
+	cluster.Name = "pgc1"
+	cluster.Spec.UserInterface =
+		&v1beta1.UserInterfaceSpec{PGAdmin: &v1beta1.PGAdminPodSpec{}}
+
+	t.Run("NoPods", func(t *testing.T) {
+		r := new(Reconciler)
+		r.Client = fake.NewClientBuilder().WithLists(new(corev1.PodList)).Build()
+		assert.NilError(t, r.reconcilePGAdminUsers(ctx, cluster, nil, nil))
+	})
+
+	// Pod labeled in the namespace
+	pod := corev1.Pod{}
+	pod.Namespace = cluster.Namespace
+	pod.Labels = map[string]string{
+		"postgres-operator.crunchydata.com/cluster": cluster.Name,
+		"postgres-operator.crunchydata.com/role":    naming.RolePGAdmin,
+	}
+
+	t.Run("PodNotRunning", func(t *testing.T) {
+		pod := pod.DeepCopy()
+		r := new(Reconciler)
+
+		// Container not running
+		pod.Status.ContainerStatuses = nil
+		r.Client = fake.NewClientBuilder().
+			WithLists(&corev1.PodList{Items: []corev1.Pod{*pod}}).
+			Build()
+
+		assert.NilError(t, r.reconcilePGAdminUsers(ctx, cluster, nil, nil))
+
+		// Pod terminating
```

Comment:
Would the "pod terminating" section test a different path in the code, since the `running` bool is still false and the condition we're looking for in pgadmin.go#330 is `running && !terminating`?

Reply:
The idea was to quickly call out both criteria, but this fails if I remove `running` from the implementation. I'll rework it.

Reply:
Split into two subtests.